libbeat/outputs/elasticsearch/client.go

```diff
@@ -61,6 +62,7 @@ func NewClient(
 	esURL, index string, proxyURL *url.URL, tls *tls.Config,
 	username, password string,
 	params map[string]string,
+	onConnectCallback *func(client *Client),
```

Comment:
onConnectCallback does not need to be a pointer to a function. Just use `func (client *Client)`. One can check for `== nil` and `!= nil` on func objects only.